dissect/target/plugins/os/unix/_os.py

```diff
@@ -301,37 +317,36 @@ def _parse_os_release(self, glob: str | None = None) -> dict[str, str]:
                                 continue
         return os_release
 
-    def _get_architecture(self, os: str = "unix", path: str = "/bin/ls") -> str | None:
-        arch_strings = {
-            0x00: "Unknown",
-            0x02: "SPARC",
-            0x03: "x86",
-            0x08: "MIPS",
-            0x14: "PowerPC",
-            0x16: "S390",
-            0x28: "ARM",
-            0x2A: "SuperH",
-            0x32: "IA-64",
-            0x3E: "x86_64",
-            0xB7: "AArch64",
-            0xF3: "RISC-V",
-        }
-
-        for fs in self.target.filesystems:
-            if fs.exists(path):
-                fh = fs.open(path)
-                fh.seek(4)
-                # ELF - e_ident[EI_CLASS]
-                bits = unpack("B", fh.read(1))[0]
-                fh.seek(18)
-                # ELF - e_machine
-                arch = unpack("H", fh.read(2))[0]
-                arch = arch_strings.get(arch)
-
-                if bits == 1:  # 32 bit system
-                    return f"{arch}_32-{os}"
-                else:
-                    return f"{arch}-{os}"
+    def _get_architecture(self, os: str = "unix", path: TargetPath | str = "/bin/ls") -> str | None:
+        """Determine architecture by reading an ELF header of a binary on the target."""
+
+        if not isinstance(path, TargetPath):
+            CANDIDATE_FOUND = False
+            for fs in self.target.filesystems:
+                if fs.exists(path):
+                    path = fs.path(path)
+                    CANDIDATE_FOUND = True
+                    break
+
+            if not CANDIDATE_FOUND:
+                return
+
+        if not path.exists():
+            return
+
+        fh = path.open("rb")
+        fh.seek(4)
+        # ELF - e_ident[EI_CLASS]
+        bits = unpack("B", fh.read(1))[0]
+        fh.seek(18)
+        # ELF - e_machine
+        arch = unpack("H", fh.read(2))[0]
+        arch = ARCH_MAP.get(arch)
+
+        if bits == 1:  # 32 bit system
+            return f"{arch}_32-{os}"
+        else:
+            return f"{arch}-{os}"
```

Comment:
Would this work instead?
```suggestion
        if not isinstance(path, TargetPath):
            path = self.target.fs.path(path)
        
        if (path := self.target.fs.path(path)).exists():
            fh = path.open("rb")
            fh.seek(4)
            # ELF - e_ident[EI_CLASS]
            bits = unpack("B", fh.read(1))[0]
            fh.seek(18)
            # ELF - e_machine
            arch = unpack("H", fh.read(2))[0]
            arch = ARCH_MAP.get(arch)

            if bits == 1:  # 32 bit system
                return f"{arch}_32-{os}"
            else:
                return f"{arch}-{os}"
```

Reply:
Keeping the loop in should be fine, but it can be written a bit nicer without the `CANDIDATE_FOUND` and more Pythonic with `None` checks. Maybe introduce a new variable `pathobj` (or swapping it around and introduce a `pathstr`, set `path` to `None` and check for a `if path is None or not path.exists()`)?

The loop should also probably start with `self.target.fs` as the first filesystem to try and check, followed by the others. I'm not sure if our "custom list" at `.filesystems` supports the `[self.target.fs, *self.target.filesystems]` syntax though. The check in the loop can also be simplified with a `fs.path(path).exists()`.